persistence/sql/persister_recovery.go

```diff
@@ -127,3 +128,67 @@ func (p *Persister) DeleteExpiredRecoveryFlows(ctx context.Context, expiresAt ti
 	}
 	return nil
 }
+
+func (p *Persister) CreateRecoveryCode(ctx context.Context, recoveryCode *code.RecoveryCode) error {
+	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.CreateRecoveryCode")
+	defer span.End()
+
+	code := recoveryCode.Code
+	recoveryCode.Code = p.hmacValue(ctx, code)
+	recoveryCode.NID = p.NetworkID(ctx)
+
+	// This should not create the request eagerly because otherwise we might accidentally create an address that isn't
+	// supposed to be in the database.
+	if err := p.GetConnection(ctx).Create(recoveryCode); err != nil {
+		return err
+	}
+
+	recoveryCode.Code = code
+	return nil
+}
+
+func (p *Persister) UseRecoveryCode(ctx context.Context, fID uuid.UUID, codeVal string) (*code.RecoveryCode, error) {
+	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.UseRecoveryCode")
+	defer span.End()
+
+	var recoveryCode code.RecoveryCode
+
+	nid := p.NetworkID(ctx)
+	if err := sqlcon.HandleError(p.Transaction(ctx, func(ctx context.Context, tx *pop.Connection) (err error) {
+		for _, secret := range p.r.Config().SecretsSession(ctx) {
```

Comment:
I suggest to use the generic system here instead. The session secret is used to sign cookies :)

Reply:
Not sure what you mean by that. This is copied from the `UseRecoveryToken` method. Could you point me to a better solution?

Reply:
I have refactored this quite a bit, after @zepatrik pointed out some improvements here. The initial question still stands though, as I don't know which secrets we can use instead.